src/actions/actions/services/dispatchers/verusid/verusid.js

```diff
@@ -56,3 +70,253 @@ export const unlinkVerusId = async (iAddress, chain) => {
     state.authentication.activeAccount.accountHash,
   );
 };
+
+export const setRequestedVerusId = async (iAddress, provisioningDetails, chain) => {
+  const state = store.getState();
+
+  if (state.authentication.activeAccount == null) {
+    throw new Error('You must be signed in for ID provisioning');
+  }
+
+  const serviceData = await requestServiceStoredData(VERUSID_SERVICE_ID);
+  const currentPendingIdentities =
+    serviceData.pending_ids == null ? {} : serviceData.pending_ids;
+
+  return await modifyServiceStoredDataForUser(
+    {
+      ...serviceData,
+      pending_ids: {
+        ...currentPendingIdentities,
+        [chain]: currentPendingIdentities[chain]
+          ? {
+            ...currentPendingIdentities[chain],
+            [iAddress]: provisioningDetails,
+          }
+          : { [iAddress]: provisioningDetails },
+      },
+    },
+    VERUSID_SERVICE_ID,
+    state.authentication.activeAccount.accountHash,
+  );
+};
+
+export const deleteProvisionedIds = async (iAddress, chain) => {
+  const state = store.getState();
+
+  if (state.authentication.activeAccount == null) {
+    throw new Error('You must be signed in for ID provisioning');
+  }
+
+  const serviceData = await requestServiceStoredData(VERUSID_SERVICE_ID);
+  const currentPendingIdentities =
+    serviceData.pending_ids == null ? {} : serviceData.pending_ids;
+
+  if (currentPendingIdentities[chain]) {
+    delete currentPendingIdentities[chain][iAddress];
+  }
+
+  return await modifyServiceStoredDataForUser(
+    {
+      ...serviceData,
+      pending_ids: currentPendingIdentities,
+    },
+    VERUSID_SERVICE_ID,
+    state.authentication.activeAccount.accountHash,
+  );
+};
+
+export const deleteAllProvisionedIds = async () => {
+  const state = store.getState();
+
+  if (state.authentication.activeAccount == null) {
+    throw new Error('You must be signed in for ID provisioning');
+  }
+
+  const serviceData = await requestServiceStoredData(VERUSID_SERVICE_ID);
+
+  return await modifyServiceStoredDataForUser(
+    {
+      ...serviceData,
+      pending_ids: {},
+    },
+    VERUSID_SERVICE_ID,
+    state.authentication.activeAccount.accountHash,
+  );
+};
+
+export const checkVerusIdNotificationsForUpdates = async () => {
+  const state = store.getState();
+
+  const getPotentialPrimaryAddresses = async (coinObj, channel) => {
+    const seeds = await requestSeeds();
+
+    const seed = seeds[channel];
+
+    const keyObj = await deriveKeyPair(seed, coinObj, channel);
+    const {addresses} = keyObj;
+
+    return addresses;
+  };
+
+  if (state.authentication.activeAccount == null) {
+    throw new Error('You must be signed in for ID provisioning functions');
+  }
+  const isTestnet = Object.keys(state.authentication.activeAccount.testnetOverrides).length > 0;
```

Comment:
This function relies on activeAccount not being null, so we can rely on activeAccount.keys existing. This means it would be preferable to get addresses from activeAccount.keys[coinObj][channel].addresses instead of accessing the seeds themselves.